packages/upload-client/src/index.js

```diff
@@ -221,3 +217,19 @@ async function uploadBlockStream(
 
   return root
 }
+
+/**
+ * Indexes a sharded DAG
+ *
+ * @param {import('multiformats').Link} root
+ * @param {import('./types.js').CARLink[]} shards
+ * @param {Array<Map<import('./types.js').SliceDigest, import('./types.js').Position>>} shardIndexes
+ */
+export async function indexShardedDAG(root, shards, shardIndexes) {
```

Comment:
This should be an export from the `blob-index` utils.

Reply:
addressed